src/lib/model/stateMachine/ConnectionStateMachine.ts

```diff
@@ -91,10 +98,15 @@ class ConnectionStateMachine {
   }
 
   public async start() {
-    // TODO: uncomment the following 2 lines when the bug https://github.com/statelyai/xstate/issues/871 is fixed
-    // const state = await this.opts.vault.getStateMachineState();
-    // this.service.start(state);
-    this.service.start();
+    const state = await this.opts.vault.getStateMachineState();
```

Comment:
Clarification:

Is the state-machine state that is stored in vault a single state for all processes?

Reply:
Yes, it's a single state machine.